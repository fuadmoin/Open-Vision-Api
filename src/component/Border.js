import React from 'react';

const Border = () => (
  <div>Border</div>
);

export default Border;
