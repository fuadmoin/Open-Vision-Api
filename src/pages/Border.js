import React from 'react';
import { useSelector } from 'react-redux';
import Details from '../component/Details';
import picture from '../img/border2.png';

const Border = () => {
  const { animals } = useSelector((state) => state.animal);
  const category2 = animals.filter((animal) => animal.breeds.primary === 'Domestic Short Hair' || animal.breeds.primary === 'Border'
  || animal.breeds.primary === 'German Shorthaired Pointer' || animal.breeds.primary === 'Golden Retriever' || animal.breeds.primary === 'Australian Cattle Dog');

  return (
    <div>
      <Details picture={picture} animals={category2} title="Border" />
    </div>
  );
};

export default Border;
