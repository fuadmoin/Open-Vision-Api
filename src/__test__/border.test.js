import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import Border from '../pages/Border';


const mockStore = configureStore([]);

describe('Border', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      animal: {
        animals: [
          { breeds: { primary: 'Pug' } },
          { breeds: { primary: 'Border' } },
          { breeds: { primary: 'Labrador Retriever' } },
        ],
      },
    });

    component = render(
      <Provider store={store}>
        <BrowserRouter>
        <Border />
        </BrowserRouter>
      </Provider>
    );
  });

  it('should render the Border component', () => {
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('should display the breed "border"', () => {
    expect(screen.getAllByText(/Border/i)[0]).toBeInTheDocument();
  });
});