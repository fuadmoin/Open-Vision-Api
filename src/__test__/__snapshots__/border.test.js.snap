// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Border should render the Border component 1`] = `
<DocumentFragment>
  <div>
    <div>
      <nav
        class="header"
      >
        <div
          class="goBack"
        >
           
          <a
            aria-current="page"
            class="active"
            href="/"
            style="text-decoration: none; color: black; grid-column: 1 / span 2;"
          >
            <img
              alt="back"
              src="left.png"
            />
            <span
              class="homeLink2"
            >
              Home
            </span>
             
          </a>
        </div>
        <span
          class="headerTitle2"
        >
          Border
        </span>
        <div
          class="headerRight2"
        >
          <img
            alt="mic"
            class="setting2"
            src="mic.png"
          />
          <img
            alt="setting"
            class="setting2"
            src="settings.png"
          />
        </div>
      </nav>
      <main
        class="detailsContainer"
      >
        <header
          class="item"
          style="background: url(border2.png) no-repeat center; background-color: rgb(230, 74, 131); height: 200px; color: rgb(233, 230, 230);"
        >
           
          <small
            class="allPets"
          >
             ANIMALS AND THEIR BREEDS  
          </small>
          <small
            class="numbe"
          >
             1
            <small>
               animals
            </small>
          </small>
        </header>
        <small
          class="title"
        >
          INDIVIDUAL BREEDS
        </small>
        <section
          class="breedSection"
        >
          <article
            class="breedsContainer"
          >
            <small
              class="breeds"
            >
              Border
            </small>
            <div
              class="rightSection"
            >
              <small
                class="numbe nums"
              >
                 1
                <small
                  class="views"
                >
                   animal
                </small>
              </small>
              <img
                alt="right"
                class=" right2"
                src="right.png"
              />
            </div>
          </article>
        </section>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
